pkg/subctl/cmd/diagnose/globalnet.go

```diff
@@ -0,0 +1,254 @@
+
+		if numberOfIPs != len(gip.Status.AllocatedIPs) {
+			status.Failure("The number of requested IPs (%d) does not match the number allocated (%d) for GlobalEgressIP %q",
+				numberOfIPs, len(gip.Status.AllocatedIPs), gip.Name)
+
+			retValue = false
+		}
+	}
+
+	return retValue
+}
+
+func checkGlobalIngressIps(cluster *cmd.Cluster, status reporter.Interface) bool {
+	mcsClient, err := mcsClientset.NewForConfig(cluster.Config)
+	if err != nil {
+		status.Warning("Error obtaining mcs client: %v", err)
+		return false
+	}
+
+	serviceExports, err := mcsClient.MulticlusterV1alpha1().ServiceExports(corev1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		status.Warning("Error listing ServiceExport resources: %v", err)
+		return false
+	}
+
+	retValue := true
+
+	for i := range serviceExports.Items {
+		ns := serviceExports.Items[i].GetNamespace()
+		name := serviceExports.Items[i].GetName()
+
+		svc, err := cluster.KubeClient.CoreV1().Services(ns).Get(context.TODO(), name, metav1.GetOptions{})
+
+		if apierrors.IsNotFound(err) {
+			status.Warning("No matching Service resource found for exported service \"%s/%s\"", ns, name)
+			continue
+		}
+
+		if err != nil {
+			status.Failure("Error retrieving Service \"%s/%s\", %v", ns, name, err)
+
+			retValue = false
+
+			continue
+		}
+
+		if svc.Spec.Type != corev1.ServiceTypeClusterIP {
+			continue
+		}
+
+		globalIngress, err := cluster.SubmClient.SubmarinerV1().GlobalIngressIPs(ns).Get(context.TODO(), name, metav1.GetOptions{})
+
+		if apierrors.IsNotFound(err) {
+			status.Failure("No matching GlobalIngressIP resource found for exported service \"%s/%s\"", ns, name)
+
+			retValue = false
+
+			continue
+		}
+
+		if err != nil {
+			status.Failure("Error retrieving GlobalIngressIP for exported service \"%s/%s\": %v", ns, name, err)
+			return false
+		}
+
+		if globalIngress.Status.AllocatedIP == "" {
+			status.Failure("No global IP was allocated for the GlobalIngressIP associated with exported service \"%s/%s\"", ns, name)
+
+			retValue = false
+
+			continue
+		}
+
+		svcs, err := cluster.KubeClient.CoreV1().Services(ns).List(
+			context.TODO(), metav1.ListOptions{LabelSelector: fmt.Sprintf("submariner.io/exportedServiceRef=%s", name)})
+		if err != nil {
+			status.Failure("Error listing internal Services \"%s/%s\": %v", ns, name, err)
+
+			retValue = false
+
+			continue
+		}
+
+		if len(svcs.Items) == 0 {
+			status.Failure("No internal service found for exported service \"%s/%s\"", ns, name)
+
+			retValue = false
+
+			continue
+		}
+
+		if len(svcs.Items) > 1 {
+			status.Failure("Found %d internal services for exported service \"%s/%s\" - expected 1", len(svcs.Items), ns, name)
+
+			retValue = false
+
+			continue
+		}
+
+		if svcs.Items[0].Spec.ExternalIPs[0] != globalIngress.Status.AllocatedIP {
+			status.Failure(
+				"The external IP (%s) for internal svc associated with exported svc \"%s/%s\" doesn't match allocated IP (%s) in GlobalIngressIP %q",
```

Comment:
```suggestion
				"The external IP (%s) for internal svc associated with exported service \"%s/%s\" doesn't match allocated IP (%s) in GlobalIngressIP %q",
```

Reply:
Yeah, I changed to 'svc' because I didn't find how to split long string into multiple lines (to avoid lint error)
Any idea how to tackle that?

Reply:
You can use + to split it, eg

```
"The external IP (%s) for internal svc associated with exported service \"%s/%s\" doesn't match " +
"allocated IP (%s) in GlobalIngressIP %q"

```